pkg/settings/source/local/local.go

```diff
@@ -41,6 +42,6 @@ func (*LocalSource) GetSourceName() string {
 }
 
 //GetSettings get settings given the key
-func (lSource *LocalSource) GetSettings(key string) (*global.Settings, error) {
+func (lSource *LocalSource) GetSettings(r *http.Request) (*global.Settings, error) {
```

Comment:
What if instead of `http.Request` we used something like `map[string]string` instead? i.e. a map of key/value pairs that we would reference when retrieving a particular setting.

Reply:
We discussed this yesterday, I agree this isn't a great option but the time investment required to come up with a better solution isn't the priority right now. This is a result of the key being contained as a header in the request but that header won't be standard across different multi-tenant implementations so we can't assume the header name in the OSS implementation.